### DISCUSSION TITLE
Static Website for this project - discussions

### DISCUSSION THREAD

**Discussion by avinal** [HarshCasper/Rotten-Scripts#755] [Ideas]
**Given below is the summary from the discussion on issue #656 .**
It is decided to create an SSG site based on Sphinx and with Sphinx Book Theme. Under this project, all the existing `markdown` files will be converted to `restructuredtext`. Although sphinx can work with markdown files, restructuredText offers much powerful and rich formatting and many options that markdown lacks, also sphinx works natively and better with restructuredtext. 

On the other side, GitHub also supports  `rst` files for READMEs so it not gonna create problems in repositories. The site can be easily hosted on GitHub Pages.

## Progress
- [x] Conversion of markdown to restructuredtext
- [ ] Removal of ambiguities and adding sphinx based directives (*in progress*)
- [ ] Page publishing
- [ ] Feedback and fixing bugs, broken links, and images
- [ ] More modifications and improvements
- [ ] Add plugins and features like SEO, binder integration etc.
- [ ] Finalise website for general use 

## Tech stacks
- Sphinx
- Sphinx Book Theme
- reStructuredText
- GitHub Actions
- GitHub Pages

**Contributions are open, please share your ideas/suggestions below** (currently at https://github.com/avinal/rotten-scripts and hosted at https://avinal.space/Rotten-Scripts)

**Comment by HarshCasper** [comment]
Thanks for all the brilliant work and effort you have put forward to it!

On a second note, the Script Winter of Code is already over and the Leaderboard has been added to the Program Admins. Please let me know if you are willing to work on this, even after the same. 

You can participate in more OSS Programs that this Project is a part of like [Cross Winter of Code](https://docs.google.com/forms/d/e/1FAIpQLSf6KUBBXKTANSWqReVakRirrNTwe10JwsMLnRpc5YygF2E5BQ/viewform) and [Mexili Winter of Code](https://mexili.github.io/winter_of_code/#/winter_of_code). 

Also to better ensure that the scores and tracked, we can break this down into multiple issues and create a Project Board for the same. let me know your thoughts on the same.

**Reply by avinal** [reply]
Thank you Harsh,
If I am not mistaken, SWOC is supposed to end on 28th February. I was hoping to get some more points 😅. Anyway, I just did it out of curiosity, so doesn't matter much. And I may not participate in more programs, but I will surely complete the website.

For the website part, I think most of the website is almost complete. I made great use of scripting to automate most of the tasks. The only thing I am unable to do/ won't be doing is, I see many of the Readme pages are mostly blank or have a very little description(mostly python). So we may want to create issues to improve them(at least description, working, output).  

I know this is not a good coding practice to merge such large changes, but I can assure you, I am checking each file individually and improving as much as I can, and following one commit per script, this thing is taking time, otherwise website is ready to publish.

**Reply by HarshCasper** [reply]
Yeah! 

It is going to end on 28th February, but the Program Admins are freezing the leaderboard right now. I am in touch with them about the same and will see how much flexible we can get there. 

You may create Issues for the Scripts that lack a proper doc as of now. Either I will solve them or will ask the first-time contributors to plug the differences.

**Comment by avinal** [comment]
I have filtered out a list of incomplete README(s) on a threshold of 30 lines at least. Here is the list, It may be put into a different issue to resolve once the website is merged. 

## Incomplete Readmes
- [ ] Go/GIF_Creator/README.rst
- [ ] JavaScript/Amazon_Review_Scrapper/README.rst
- [ ] JavaScript/Automate_Tinder/README.rst
- [ ] JavaScript/Base64_Encoder_Decoder/README.rst
- [ ] JavaScript/Bulk_LinkedIn_requests/README.rst
- [ ] JavaScript/Edit_Webpage/README.rst
- [ ] JavaScript/Follow_Instagram/README.rst
- [ ] JavaScript/Html_to_Markdown/README.rst
- [ ] JavaScript/Instagram_Like/README.rst
- [ ] JavaScript/Send_Emails/README.rst
- [ ] JavaScript/Who_Don't_Follow_You/README.rst
- [ ] Python/All_GitHub_Repos/README.rst
- [ ] Python/Book_Downloader/README.rst
- [ ] Python/Codechef_Contests/README.rst
- [ ] Python/CRICBUZZ_scraper/README.rst
- [ ] Python/cryptocurrency_converter/README.rst
- [ ] Python/Currency_Converter/README.rst
- [ ] Python/Download_Instagram_Posts/README.rst
- [ ] Python/Download_Instagram_Videos/README.rst
- [ ] Python/DSC_Event_Scrapper/README.rst
- [ ] Python/Email_Scrape_From_Webpage/README.rst
- [ ] Python/Encrypt_Text/README.rst
- [ ] Python/Extract_Image_Metadata/README.rst
- [ ] Python/File_Organiser/README.rst
- [ ] Python/Find_All_Links/README.rst
- [ ] Python/Flask_Directory_Initiator/README.rst
- [ ] Python/Generate_Fake_Data/README.rst
- [ ] Python/Get_Lyrics/README.rst
- [ ] Python/Google-News-Scrapper/README.rst
- [ ] Python/Heat_Your_System/README.rst
- [ ] Python/HTML_to_Markdown/README.rst
- [ ] Python/Image_Encoder/README.rst
- [ ] Python/Image_flip_meme_templates/README.rst
- [ ] Python/Image_Sketch/README.rst
- [ ] Python/Image_to_Sepia/README.rst
- [ ] Python/IMDb_Top_Movies/README.rst
- [ ] Python/Instagram_Followers_Following_Scraper/README.rst
- [ ] Python/Instagram_Followers_Following_Scraper/Using_BeautifulSoup/README.rst
- [ ] Python/Internshala_Scrapper/README.rst
- [ ] Python/Invisible_Cloak/README.rst
- [ ] Python/JSON_to_Excel/README.rst
- [ ] Python/Labelling_Binarizer_Instance/README.rst
- [ ] Python/Large_File_Identifier/README.rst
- [ ] Python/Lorem_Ipsum_Generator/README.rst
- [ ] Python/Magic_Pen/README.rst
- [ ] Python/markovmemes/README.rst
- [ ] Python/Meet_Strangers/README.rst
- [ ] Python/Net_Connection/README.rst
- [ ] Python/OCR_To_Google_Question_Papers/README.rst
- [ ] Python/Python_Library_Downloader/README.rst
- [ ] Python/Query_Links_Script/README.rst
- [ ] Python/Random_Password_Generator/README.rst
- [ ] Python/Reddit_Wallpaper/README.rst
- [ ] Python/ROT13/README.rst
- [ ] Python/S3_Bucket_Downloader/README.rst
- [ ] Python/Scrape_Twitter/README.rst
- [ ] Python/Screen_Grabber/README.rst
- [ ] Python/Search_File/README.rst
- [ ] Python/Send_Emails/README.rst
- [ ] Python/Shutdown_Your_System/README.rst
- [ ] Python/Speech_to_Text/README.rst
- [ ] Python/Story_Reader/README.rst
- [ ] Python/Strong_Password_Detection/README.rst
- [ ] Python/Substitution_Cypher/README.rst
- [ ] Python/Test_Typing_Speed/README.rst
- [ ] Python/Text_To_Speech/README.rst
- [ ] Python/Times_Of_India/README.rst
- [ ] Python/TSV_to_CSV/README.rst
- [ ] Python/URL_Shortner/README.rst
- [ ] Python/Vigenere_Cipher/README.rst
- [ ] Python/Watermark_it/README.rst
- [ ] Python/Weather_using_OpenWeatherMap/README.rst
- [ ] Python/Website_Cloner/README.rst
- [ ] Python/Who_Is_Online_On_WhatsApp/README.rst
- [ ] Python/Wifi_Speed_Tester/README.rst
- [ ] Python/Word_Meaning/README.rst
- [ ] Python/XML_To_CSV/README.rst
- [ ] Python/YouTube_Video-Audio_Download/README.rst
- [ ] Python/Youtube_Video_Download/README.rst
- [ ] Python/Zipper/README.rst

P.S - Some may be complete, I haven't checked them again one-on-one but according to the template, at least 25-30 lines must be there.

**Reply by HarshCasper** [reply]
Thanks for this! 

I will make them separate Issues that first-time contributors can work on.

**Reply by avinal** [reply]
Can we ask people to do it directly in restructruredtext? then we won't need to update them again. I can write a template and a small guide to follow, only a few directives will be needed so not very hard to catch up. (reference #766 )

**Reply by HarshCasper** [reply]
That's a good idea. What do you say @vybhav72954 @seema1711 ?

**Reply by vybhav72954** [reply]
@avinal Am sorry but I can't follow what you mean by restructured text though.

Update - I am assuming it is by that you mean the branch via which you pushed this website.

Proposal - @avinal PLease make a new Template for this

Meanwhile, I will ensure that all the scripts are working and styled properly.

It Will take me a day, so by the day after tomorrow, I will push the same.

After that according to your template, I will make the changes in the READMEs as mentioned in the list.

Then I will push the PR for the same (updated scripts and the updated README), in the required branch. This will take some time, but I think we together can do this.

**Reply by avinal** [reply]
restructuredtext is nothing new, just as markdown, it is a text formatting syntax, but offers a much powerful syntax and a vast variety of options/formats, please wait for some time, I will add a small guide that will help you catch up easily.

**Reply by avinal** [reply]
@vybhav72954 Here are some tips for rst syntax. https://www.avinal.space/posts/development/rst-guide.html

It would be all that you will need, in case you want richer formatting or more options. See here: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html

**Please keep in mind that, restructuredtext is indentation sensitive, just as python, so spaces and tabs behave differently, also 1 tab must be exactly 4 spaces.** If you use VS Code then you can use [this](https://github.com/vscode-restructuredtext/vscode-restructuredtext) extension to make your work a lot easier.

I have also attached the template. Tell me if you have any more doubts.

[TEMPLATE.txt](https://github.com/HarshCasper/Rotten-Scripts/files/5978131/TEMPLATE.txt)

**Comment by vybhav72954** [comment]
Hey @avinal I believe you might be using Discord, can you tell me your tag or maybe shoot me a DM at `Astrian#3019`

So that we can discuss the Website and proceed accordingly, let's try to finalize this asap so that we can launch this with a bang.

**Reply by avinal** [reply]
Yeah Sure

**Comment by avinal** [comment]
Hello @HarshCasper, it's been a while. I think it will be wise to mandate new PRs for README changes in restructuredtext instead of markdown. There are many options we can do to further simplify the process:-
- As I have directed this PR to `rotten-script-website` instead of `main` you can merge this and accept new changes in this branch(by making it default), especially for issue #656 
- Or mandate new PRs in restructuredtext and later realign everything(still needs to be rebased with this PR someday)
- Complete the review soon and merge it to `main`, with 2nd option (risky but everything is double-checked from my side and none of the actual scripts has been changed, plus it will allow website publication simultaneously, only issues are README and they are being covered in issues)
- propose something else soon because programs will come and go and with every new change this PR becomes less relevent and won't be much useful someday (a large work lost😥)
I hope we will soon come up with plans. Tagging @vybhav72954 for suggestions

**Reply by vybhav72954** [reply]
> Hello @HarshCasper, it's been a while. I think it will be wise to mandate new PRs for README changes in restructuredtext instead of markdown. There are many options we can do to further simplify the process:-
> 
> * As I have directed this PR to `rotten-script-website` instead of `main` you can merge this and accept new changes in this branch(by making it default), especially for issue #656
> * Or mandate new PRs in restructuredtext and later realign everything(still needs to be rebased with this PR someday)
> * Complete the review soon and merge it to `main`, with 2nd option (risky but everything is double-checked from my side and none of the actual scripts has been changed, plus it will allow website publication simultaneously, only issues are README and they are being covered in issues)
> * propose something else soon because programs will come and go and with every new change this PR becomes less relevent and won't be much useful someday (a large work lost😥)
>   I hope we will soon come up with plans. Tagging @vybhav72954 for suggestions

@HarshCasper Let's have a meeting within a day or two. We need to discuss this. Moreover, I suggest not to accept any new PR before we come to a solution to this

**Reply by avinal** [reply]
Yepp, let us have a meeting soon.
